crates/ra_hir/src/code_model_api.rs

```diff
@@ -272,6 +278,10 @@ impl Struct {
     pub fn generic_params(&self, db: &impl HirDatabase) -> Arc<GenericParams> {
         db.generic_params((*self).into())
     }
+
+    pub fn ty(&self, db: &impl HirDatabase) -> Ty {
+        type_for_struct(db, *self)
```

Comment:
I think this could be `db.type_for_def((*self).into())`, and the same for `enum`. So, no need to make `type_for_struct` more public.